src/aws/osml/image_processing/gdal_tile_factory.py

```diff
@@ -135,7 +139,199 @@ def create_encoded_tile(
+            src_dim = np.min([src_bbox[2] - src_bbox[0], src_bbox[3] - src_bbox[1]])
+            return int(np.max([0, int(np.floor(np.log2(src_dim / tile_width)))]))
+
+        num_overviews = self.raster_dataset.GetRasterBand(1).GetOverviewCount()
+        r_level = min(find_appropriate_r_level(src_bbox, tile_size[0]), num_overviews)
+        overview_bbox = tuple([int(x / 2**r_level) for x in src_bbox])
+        logger.debug(f"Using r-level: {r_level}")
+        logger.debug(f"overview_bbox = {overview_bbox}")
+        logger.debug(f"Dataset size = {self.raster_dataset.RasterXSize},{self.raster_dataset.RasterYSize}")
+
+        # Read pixels from the selected resolution level that match the region of the image needed to create the
+        # map tile. This data becomes the "src" in the cv2.remap transformation.
+        src = self._read_from_rlevel_as_array(overview_bbox, r_level)
+        logger.debug(f"src.shape = {src.shape}")
+
+        # Update the src_x and src_y coordinates because we cropped the image and pulled it from a different
+        # resolution level. The original coordinates assumed the image origin at 0,0 in a full resolution
+        # image
+        src_x = (src_x - src_bbox[0]) / 2**r_level
+        src_y = (src_y - src_bbox[1]) / 2**r_level
+
+        # Create 2D linear interpoloators that map the pixels in the map tile to x and y values in the source image.
+        # This will allow us to efficiently generate the maps needed by the opencv::remap function for every pixel
+        # in the destination image.
+        src_x_interpolator = RectBivariateSpline(pixel_x, pixel_y, src_x, kx=1, ky=1)
+        src_y_interpolator = RectBivariateSpline(pixel_x, pixel_y, src_y, kx=1, ky=1)
+
+        # Create the map1 and map2 arrays that capture the non-linear relationship between each pixel in the map tile
+        # (dst) to pixels in the original image (src). See opencv::remap documentation for definitions of these
+        # parameters.
+        dst_x = np.linspace(0, tile_size[0] - 1, tile_size[0])
+        dst_y = np.linspace(0, tile_size[1] - 1, tile_size[1])
+        map1 = src_x_interpolator(dst_x, dst_y).astype(np.float32)
+        map2 = src_y_interpolator(dst_x, dst_y).astype(np.float32)
+
+        logger.debug(
+            f"Sanity check remap array sizes. " f"They should match the desired map tile size {tile_size[0]}x{tile_size[1]}"
+        )
+        logger.debug(f"map1.shape = {map1.shape}")
+        logger.debug(f"map2.shape = {map2.shape}")
+
+        dst = cv2.remap(src, map1, map2, cv2.INTER_LINEAR)
+        output_tile_pixels = self._create_display_image(dst)
+
+        # TODO: Formats other than PNG?
+        is_success, image_bytes = cv2.imencode(".png", output_tile_pixels)
+        if is_success:
+            return image_bytes
+        else:
+            return None
+
+    def _read_from_rlevel_as_array(
+        self, scaled_bbox: tuple[int, int, int, int], r_level: int, band_numbers: List[int] = None
```

Comment:
You can make this List[int] = None Optional instead to match convention.

Reply:
Yep. Will fix.